dotnet/samples/KernelSyntaxExamples/Example23_OpenAPIPlugin.cs

```diff
@@ -0,0 +1,101 @@
+        Kernel kernel = new();
+
+        // This HTTP client is optional. SK will fallback to a default internal one if omitted.
+        using HttpClient httpClient = new();
+
+        // Create a sample OpenAPI schema that calls the github versions api, and has an operation extension property.
+        // The x-openai-isConsequential property is the operation extension property.
+        var schema =
+            "{" +
+            "  \"openapi\": \"3.0.1\"," +
+            "  \"info\": {" +
+            "    \"title\": \"Github Versions API\"," +
+            "    \"version\": \"1.0.0\"" +
+            "  }," +
+            "  \"servers\": [" +
+            "    {" +
+            "      \"url\": \"https://api.github.com\"" +
+            "    }" +
+            "  ]," +
+            "  \"paths\": {" +
+            "    \"/versions\": {" +
+            "      \"get\": {" +
+            "        \"x-openai-isConsequential\": false," +
+            "        \"operationId\": \"getVersions\"," +
+            "        \"responses\": {" +
+            "          \"200\": {" +
+            "            \"description\": \"OK\"" +
+            "          }" +
+            "        }" +
+            "      }" +
+            "    }" +
+            "  }" +
+            "}";
+        var schemaStream = new MemoryStream();
+        WriteStringToStream(schemaStream, schema);
+
+        // Import an Open AI plugin from a stream.
+        var plugin = await kernel.CreatePluginFromOpenApiAsync("GithubVersionsApi", schemaStream, new OpenAIFunctionExecutionParameters(httpClient));
+
+        // Get the function to be invoked and its metadata and extension properties.
+        var function = plugin["getVersions"];
+        function.Metadata.TryGetValue("operation-extensions", out var extensionsObject);
+        var operationExtensions = extensionsObject as Dictionary<string, object?>;
+
+        // Here is a sample use case, where we would invoke the function only if it is consequence free based on the x-openai-isConsequential extension value.
+        if (operationExtensions is null || !operationExtensions.TryGetValue("x-openai-isConsequential", out var isConsequential) || isConsequential is null)
+        {
+            WriteLine("We cannot determine if the function has consequences, since the isConsequential extension is not provided, so safer not to run it.");
+        }
+        else if ((isConsequential as bool?) == true)
+        {
+            WriteLine("This function may have unwanted consequences, so safer not to run it.");
+        }
+        else
+        {
+            // Invoke the function and output the result.
+            var functionResult = await kernel.InvokeAsync(function, new KernelArguments());
+            var result = functionResult.GetValue<RestApiOperationResponse>();
+            WriteLine($"Function execution result: {result?.Content}");
+        }
+    }
+
+    public Example23_OpenAPIPlugin(ITestOutputHelper output) : base(output)
```

Comment:
nit: This constructor can be converted to primary constructor.

Reply:
Makes sense.  Will fix.